superset/sqllab/command.py

```diff
@@ -94,11 +96,19 @@ def run(  # pylint: disable=too-many-statements,useless-suppression
                 status = SqlJsonExecutionStatus.QUERY_ALREADY_CREATED
             else:
                 status = self._run_sql_json_exec_from_scratch()
+
+            self._execution_context_convertor.set_payload(
+                self._execution_context, status
+            )
+
+            # save columns into metadata_json
+            self._query_dao.save_metadata(
+                self._execution_context.query, self._execution_context_convertor.payload
+            )
+
             return {
                 "status": status,
-                "payload": self._execution_context_convertor.to_payload(
-                    self._execution_context, status
-                ),
+                "payload": self._execution_context_convertor.serialize_payload(),
```

Comment:
Can we maybe just let the command return raw dict here, too, and use `self.json_response` in whatever view handler the result may be used, so that json.dumps in the executor itself can be cleaned up?

Reply:
Sounds good.